### ISSUE TITLE
new refrigerated tank doesn't show up as a tank or vehicle electronic

### ISSUE THREAD

**Issue by joe9000** [CleverRaven/Cataclysm-DDA#39546]
# Describe the bug
the new [refrigerated tank](https://github.com/CleverRaven/Cataclysm-DDA/commit/4563e12f6a40b80446c6ef81a9605ff47aeda746) doesn't appear as a tank in the [e]xamine vehicle screen, nor can i fill it.

# Steps To Reproduce
1. create vehicle, refrigerated tank, a rubber hose, and a water source
2. install refrigerated tank in vehicle
3. [e]xamine vehicle: refrigerated tank does not appear in tanks section
4. [e]xamine a water source adjacent to vehicle: unable to fill refrigerated tank

# Expected behavior
be able to access refrigerated tank like an existing 60l tank installed in vehicle. 

#workaround
add "insulation": 10, to containers.json for id: metal_tank

**Comment by Maleclypse** [comment]
I'm confused because  feel like this worked though I may not have tested it thoroughly enough.  These flags don't seem like they conflict with each other, `"flags": [ "FLUIDTANK", "FRIDGE", "ENABLED_DRAINS_EPOWER" ]`.  I'll test again and see if I can figure out where this is going awry.

**Comment by Maleclypse** [comment]
Plus side, debug spawned in a solar vehicle, toolbox, and refridgerated tank.  Installed tank.  examined vehicle controls and under control electronics was a new fridge I could turn on.  Downside, still not visible as a fuel tank.

**Comment by RedShakespeare** [comment]
There appear be two problems: 
1. One vehicle part can work as a fluid tank only if it is made by an item which has type "CONTAINER" and attributes "watertight" and "seals" of which have to be both true. I tried to rewrite the item-ver fridgetank like this 

```
{
    "type": "CONTAINER",
    "looks_like": "VEHICLE_PART",
    "copy-from": "washing_machine",
    "id": "fridgetank",
    "name": { "str": "refrigerated tank" },
    "description": "A 60L refrigerated tank for keeping liquids cool.  Provides some insulation from outside weather.",
    "insulation": 2,
    "volume": "60 L",
    "weight": "48 kg",
    "contains": "60 L",
    "seals": true,
    "watertight": true,
    "qualities": [ [ "CONTAIN", 1 ] ]
  },

```
and it turned out that this vehicle part gained the ability to be refilled.

2. It looks like the flag "FRIDGE" works only by changing the temperature flag of **the space inside the vehicle part itself**, which means only if you can put items with "D"/"d" key into the vehicle part will these items get cooled down. I tested this by filling this tank with raw milk and turned on the fridge, and unfortunately after 12 hour in game waiting the milk inside this tank got rotten just the same time as milk in common plastic bottles.

So, maybe the author should add some src codes to realize "cooling down liquld in a tank", or this item should be deleted from the game.

**Comment by Maleclypse** [comment]
Right now this is waiting until pockets are implemented to see how that impacts things and to not cause additional problems for the pockets code.

**Comment by Maleclypse** [comment]
Tests suggest this issue is fixed.

**Comment by anothersimulacrum** [comment]
Fixed in #42384
